bluesky/callbacks/best_effort.py

```diff
@@ -529,6 +529,46 @@ def stop(self, doc):
         self.check_visibility()
         super().stop(doc)
 
+class BaselinePrinter(Callback):
+    def __init__(self, start_doc, file=sys.stdout):
+        # We accept a start_doc but discard it.
+        ...
+        self._baseline_enabled = True
+        self._descriptors = {}
+        self._baseline_toggle = True
+        self._file = file
+    def descriptor(self, doc):
+        # TODO Check the stream name of this descriptor using doc['name']
+        # If the stream name is 'baseline', record doc['uid'] in an instance
+        # variable for future reference.
+        ...
+        self._descriptors[doc['uid']] = doc
+
+    def event(self, doc):
+        # Check the descriptor uid of this Event using doc['descriptor'].
+        # if this matches the uid of the descriptor we have stashed in the 
+        # method above, extract the baseline readings and print them, the way
+        # we do in BestEffortCallback currently.
+        ...
+        descriptor = self._descriptors[doc['descriptor']]
```

Comment:
Then, following on my comment above about making `event` more efficient, here we just have to check

```py
if doc['descriptor'] in self._descriptors:
```

which will be faster than doing `descriptor.get('name') == 'baseline'` each time.